lib/NfcoreTemplate.groovy

```diff
@@ -230,36 +233,32 @@ class NfcoreTemplate {
     //
     // Print pipeline summary on completion
     //
-    public static void summary(workflow, params, log, fail_percent_mapped=[:], pass_percent_mapped=[:]) {
+    public static void summary(workflow, params, log, pass_mapped_reads=[:], pass_trimmed_reads=[:], pass_strand_check=[:]) {
         Map colors = logColours(params.monochrome_logs)
 
-        def total_aln_count = pass_percent_mapped.size() + fail_percent_mapped.size()
-        if (pass_percent_mapped.size() > 0) {
-            def idx = 0
-            def samp_aln = ''
-            for (samp in pass_percent_mapped) {
-                samp_aln += "    ${samp.value}%: ${samp.key}\n"
-                idx += 1
-                if (idx > 5) {
-                    samp_aln += "    ..see pipeline reports for full list\n"
-                    break;
-                }
+        def fail_mapped_count  = pass_mapped_reads.count  { key, value -> value == false }
+        def fail_trimmed_count = pass_trimmed_reads.count { key, value -> value == false }
+        def fail_strand_count  = pass_strand_check.count  { key, value -> value == false }
+        if (workflow.success) {
+            def color = colors.green
+            def status = []
+            if (workflow.stats.ignoredCount != 0) {
+                color = colors.yellow
+                status += ['with errored process(es)']
             }
-            log.info "-${colors.purple}[$workflow.manifest.name]${colors.green} ${pass_percent_mapped.size()}/$total_aln_count samples passed STAR ${params.min_mapped_reads}% mapped threshold:\n${samp_aln}${colors.reset}-"
-        }
-        if (fail_percent_mapped.size() > 0) {
-            def samp_aln = ''
-            for (samp in fail_percent_mapped) {
-                samp_aln += "    ${samp.value}%: ${samp.key}\n"
+            if (fail_mapped_count > 0 || fail_trimmed_count > 0 || fail_strand_count > 0) {
```

Comment:
Samples that fail strandedness check are not skipped, they are still processed but a warning is printed in the multiqc report. For example a run with all 6 samples with the wrong strandedness specified in the sampelsheet, the below output is printed to terminal:

`-[nf-core/rnaseq] Pipeline completed successfully with skipped sampl(es)-
-[nf-core/rnaseq] Please check MultiQC report: 6/6 samples failed strandedness check.-`

Can this be changed to only report 'skipped samp(les)' for samples that fail map or trim count and not strandedness. The message that samples failed strandedness check is sufficient.

A more helpful long-term solution would be to run a QC and strand check BEFORE the whole rnaseq run with alignment and counting is performed. This is more consistent with what researchers want in real life.